SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/ContactListener.h

```diff
@@ -57,6 +58,27 @@ class SOFA_SOFABASECOLLISION_API ContactListener : public virtual core::objectmo
 
     void handleEvent( core::objectmodel::Event* event ) override;
 
+    // Returns the number of stored contacts.
+    unsigned int getNumberOfContacts() const;
+
+    // Returns the distances between the stored contacts as a vector.
+    helper::vector<double> getDistances() const;
+
+    // Returns the full ContactsVector
+    helper::vector<const helper::vector<DetectionOutput>* > getContactsVector() const;
+
+    // Returns the contact points in the form of a vector of tuples containing two positive integers and two Vector3.
+    // The Vector3 store the X, Y, Z coordinates of the points in contact
+    // The integers specify to which collision models the points belong. (e.g. (collModel2, (3., 5., 7.), collModel1, (3.1, 5., 6.9)))
+    // TODO: replace the tuple with a struct to avoid forgetting which element refers to what.
```

Comment:
This line is exactly what I wanted to say 😁

Reply:
Planning to do it in this PR or later ?

Reply:
If it is okay with you, I would do this later.
As soon as I know how I want to bind those functions in SofaPython3.

Reply:
I am OK with it, just that there are many TODOs becoming forgotten 😵

Reply:
Ok cool! :)
I will keep it in mind! :D